zwave_js_server/model/controller/statistics.py

```diff
@@ -42,19 +42,85 @@ def nlwr(self) -> Optional[RouteStatistics]:
         return self._nlwr
 
 
-class ControllerStatisticsDataType(TypedDict):
+class ChannelRSSIDataType(TypedDict):
+    """Represent a channel RSSI data dict type."""
+
+    average: int
+    current: int
+
+
+class BackgroundRSSIDataType(TypedDict, total=False):
+    """Represent a background RSSI data dict type."""
+
+    # https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/controller/ControllerStatistics.ts#L40
+    timestamp: int  # required
+    channel0: ChannelRSSIDataType  # required
+    channel1: ChannelRSSIDataType  # required
+    channel2: ChannelRSSIDataType
+
+
+class ControllerStatisticsDataType(TypedDict, total=False):
     """Represent a controller statistics data dict type."""
 
     # https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/controller/ControllerStatistics.ts#L20-L39
-    messagesTX: int
-    messagesRX: int
-    messagesDroppedTX: int
-    messagesDroppedRX: int
-    NAK: int
-    CAN: int
-    timeoutACK: int
-    timeoutResponse: int
-    timeoutCallback: int
+    messagesTX: int  # required
+    messagesRX: int  # required
+    messagesDroppedTX: int  # required
+    messagesDroppedRX: int  # required
+    NAK: int  # required
+    CAN: int  # required
+    timeoutACK: int  # required
+    timeoutResponse: int  # required
+    timeoutCallback: int  # required
+    backgroundRSSI: BackgroundRSSIDataType
+
+
+class ChannelRSSI:
```

Comment:
Alternatively make a dataclass, if we're not going to update the data after instantiation.

```py
@dataclass
class ChannelRSSI:
    data: ChannelRSSIDataType
    average: int = field(init=False)
    current: int = field(init=False)

    def __post_init__(self) -> None:
        self.average = self.data["average"]
        self.current = self.data["current"]
```

Reply:
good idea